src/manifest.rs

```diff
@@ -167,15 +167,21 @@ impl Manifest {
                              -> Result<(), ManifestError> {
         let (ref name, ref data) = dep.to_toml();
         let ref mut manifest = self.data;
-        let entry = manifest.entry(String::from(table))
-                            .or_insert_with(|| toml::Value::Table(BTreeMap::new()));
-        match *entry {
-            toml::Value::Table(ref mut deps) => {
-                deps.insert(name.clone(), data.clone());
-                Ok(())
+
+        // TODO: replace with toml::Parser::new(table).lookup() once it lands, to handle quoted parts (e.g. targets)
```

Comment:
Can you create an issue to track this once the PR lands? I tend to forget about TODOs and FIXMEs otherwise :)